server.js

```diff
@@ -1,32 +1,210 @@
+
+connectoToDB();
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
-mongoose.Promise = Promise;
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
+const swaggerDocs = swaggerJsdoc(options);
+app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs));
+
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Seed the database
+const Cup = mongoose.model('Cup', {
+  year: Number,
+  host: String,
+  winner: String,
+  second: String,
+  third: String,
+  fourth: String,
+  goals_scored: Number,
+  teams: Number,
+  games: Number,
+  attendance: Number
+});
+
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const WorldCupApi = {
+    Welcome: 'Hi! This a World Cup Api (1930 - 2018)',
+    Routes: [{
+      "/cups": 'Get all world cup.',
+      "/cups/'number'": 'Get a World Champion with Matching ID.',
+      "/cups/winner/'country'": 'Get the result of a World Champion',
+      "/cups/year/:year": 'Get the place where the World Cup was played according to the year',
+      "/endpoints": "Get API endpoints."
+    }]
+  }
+  res.send(WorldCupApi)
+});
+
+
+/**
+ * @openapi
+ * /endpoints:
+ *   get:
+ *     description: Get API endpoints.
+ *     responses: 
+ *       200: 
+ *         description: Successful response. 
+ */
+app.get("/endpoints", (req, res) => {
+  res.send(listEndpoints(app))
+});
+
+/**
+ * @openapi
+ * /cups:
+ *   get:
+ *     description: Get all World Cup.
+ *     responses: 
+ *       200: 
+ *         description: Successful response. 
+ */
+app.get("/cups", async (req, res) => {
+  let query = req.query || {}
+  try {
+    let cups = await Cup.find(query)
+    res.json(cups)
+  } catch {
+    res.status(404).json({ error: 'Not found' });
+  }
+});
+
+/**
+ * @openapi
+ * /cups/{_id}:
+ *   get:
+ *     description: Get a World Champion with Matching ID.
+ *     parameters:
+ *       - in: path
+ *         name: _id
+ *         schema:
+ *           type: string
+ *         description: Get a World Champion with Matching ID.
+ *     responses: 
+ *       200: 
+ *         description: Successful response. 
+ */
+app.get("/cups/:_id", async (req, res) => {
+  try {
+    const cupId = await Cup.findById(req.params._id)
```

Comment:
What you could do in this part is to destructurize the _id into a variable before try ( row 140) `const { id } = req.params `which would then improve to user experience as you avoid the _ character hence `const cupId = await Cup.findById(req.params.id)` and the route to "/cups/:id"